packages/typespec-azure-core/src/rules/use-standard-integer.ts

```diff
@@ -0,0 +1,35 @@
+import { Model, createRule } from "@typespec/compiler";
+
+export const useStandardInteger = createRule({
+  name: "use-standard-integer",
+  description: "Use only types that map to int32 or int64.",
+  severity: "warning",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-core/rules/use-standard-integer",
+  messages: {
+    default: "Recommended integer types are 'int32', 'int64' or 'safeint'.",
+  },
+  create(context) {
+    return {
+      model: (model: Model) => {
+        const nonRecommendedInts = [
```

Comment:
Nit: would move out of the rule and declare as a `Set`

Reply:
Done